### PR TITLE
(test) Adding test coverage for the encounter role handler

### PR DESCRIPTION
**Pull request by gitcliff**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
Adding test coverage for the encounter role handler for handlingFieldSubmission, getting initial and prevoius values

### CODE REVIEW

**Review by samuelmale — COMMENTED** [review]
Can you use a single test suite?

File: src/submission-handlers/encounterRolesHandler.test.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  encounterDate: new Date(2020, 11, 29),
+  setEncounterDate: (value) => {},
+  encounterProvider: '2c95f6f5-788e-4e73-9079-5626911231fa',
+  setEncounterProvider: jest.fn,
+  setEncounterLocation: jest.fn,
+  encounterRole: '6d95f6f5-788e-4e73-9079-5626911231f4',
+  setEncounterRole: jest.fn,
+};
+
+const encounterContext2: EncounterContext = {
```
[the remark this answers is not in the document]
Comment (gitcliff) [reply src/submission-handlers/encounterRolesHandler.test.ts]: i have removed it

File: src/submission-handlers/encounterRolesHandler.test.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    };
+    // replay
+    const role = EncounterRoleHandler.handleFieldSubmission(field, 'Clinician', encounterContext);
+    // verify
+    expect(role).toEqual('Clinician');
+  });
+});
+
+describe('EncounterRoleHandler - getInitialValue', () => {
+  // new submission (enter mode)
```
[the remark this answers is not in the document]
Comment (gitcliff) [reply src/submission-handlers/encounterRolesHandler.test.ts]: no,,getting initial value

File: src/submission-handlers/encounterRolesHandler.test.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      id: 'encounterRole',
+      questionOptions: {
+        rendering: 'ui-select-extended',
+      },
+      validators: [],
+    };
+    // replay
+    const role = EncounterRoleHandler.handleFieldSubmission(field, 'Clinician', encounterContext);
+    // verify
+    expect(role).toEqual('Clinician');
```
[the remark this answers is not in the document]
Comment (gitcliff) [reply src/submission-handlers/encounterRolesHandler.test.ts]: thanks, i have updated this for the cases in the handler dealing with the encounterContext ie [here](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/submission-handlers/encounterRoleHandler.ts#L5) and [here ](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/submission-handlers/encounterRoleHandler.ts#L17)

**Review by samuelmale — APPROVED** [review]

**Pull request merged**